In [48]:
import pandas as pd

Vamos a importar los csv relacionados por provincia, año y trimestre. 

## DF  Internet_Accesos-por-tecnologia_y_provincia.csv

In [218]:
df1= pd.read_csv('internet_TconexionD.csv')
print(df1.info())
df1.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 866 entries, 0 to 865
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           865 non-null    object 
 1   Trimestre     865 non-null    object 
 2   Provincia     864 non-null    object 
 3   ADSL          864 non-null    float64
 4   Cablemodem    864 non-null    float64
 5   Fibra óptica  864 non-null    float64
 6   Wireless      864 non-null    float64
 7   Otros         864 non-null    float64
 8   Total         864 non-null    float64
dtypes: float64(6), object(3)
memory usage: 61.0+ KB
None


Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2022         4     Buenos Aires  325908.0   2767247.0     1499149.0   
1  2022         4  Capital Federal  138753.0   1238196.0      138548.0   
2  2022         4        Catamarca    8952.0     10998.0       47337.0   
3  2022         4            Chaco   24582.0     61265.0       47285.0   
4  2022         4           Chubut   45450.0     73923.0        9815.0   

   Wireless    Otros      Total  
0  129910.0  72451.0  4794665.0  
1    4603.0  31826.0  1551926.0  
2    1379.0   2243.0    70909.0  
3    7957.0   2574.0   143663.0  
4   30721.0   8850.0   168759.0

In [219]:
datos_nulos = df1.isnull().sum()

# Imprimir los resultados
print(datos_nulos)

Año             1
Trimestre       1
Provincia       2
ADSL            2
Cablemodem      2
Fibra óptica    2
Wireless        2
Otros           2
Total           2
dtype: int64


In [220]:
ubicacion_datos_nulos = df1[df1.isnull().any(axis=1)]

# Imprimir los resultados
print(ubicacion_datos_nulos)

     Año                                          Trimestre Provincia  ADSL  \
864  NaN                                                NaN       NaN   NaN   
865    *  Los datos provinciales no coinciden a nivel na...       NaN   NaN   

     Cablemodem  Fibra óptica  Wireless  Otros  Total  
864         NaN           NaN       NaN    NaN    NaN  
865         NaN           NaN       NaN    NaN    NaN  


In [221]:
fila = df1.loc[864]
print(fila)

Año             NaN
Trimestre       NaN
Provincia       NaN
ADSL            NaN
Cablemodem      NaN
Fibra óptica    NaN
Wireless        NaN
Otros           NaN
Total           NaN
Name: 864, dtype: object


In [222]:
valores_unicos = df1['Año'].unique()
print(valores_unicos)

['2022' '2021' '2020' '2019' '2019 *' '2018' '2017' '2016' '2015' '2014'
 nan '*']


En el analisis del dataSet me di cuenta que el año  2019 hay varias filas que tiene un *  y existe tambien un año con solo un * , este ultimo corresponde a la fila 864 la cual será eliminada por no tener ningun tipo de información.

In [223]:
filas_con_asterisco = df1[df1['Año'] == '*']

# Imprimir las filas
print(filas_con_asterisco)

    Año                                          Trimestre Provincia  ADSL  \
865   *  Los datos provinciales no coinciden a nivel na...       NaN   NaN   

     Cablemodem  Fibra óptica  Wireless  Otros  Total  
865         NaN           NaN       NaN    NaN    NaN  


Vamos a modificar el el año '2019*' por '2019'   y eliminar la fila 864


In [224]:
# Modificar las filas en la columna
df1['Año'] = df1['Año'].replace('2019 *', '2019')

#eliminar la fila 864 , ya que no tiene ningun valor .
df1 = df1.drop(864)


In [225]:
valores_unicos = df1['Trimestre'].unique()
print(valores_unicos)

['4' '3' '2' '1' '3 *' '2 *' '1 *'
 'Los datos provinciales no coinciden a nivel nacional, ya que se rincorporó información que no contien apertuta a nivel geográfico.']


Aca onservamos que hay valores en el trimestre que tienen un *. Vamos hacer le reemplazo de estos valores, en  cada caso: 3 * po 3, 2 * por 2, 1 * por 1.

In [226]:
df1['Trimestre'] = df1['Trimestre'].replace('3 *','3')
df1['Trimestre'] = df1['Trimestre'].replace('2 *','2')
df1['Trimestre'] = df1['Trimestre'].replace('1 *','1')

In [203]:
df1.head()

Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2022         4     Buenos Aires  325908.0   2767247.0     1499149.0   
1  2022         4  Capital Federal  138753.0   1238196.0      138548.0   
2  2022         4        Catamarca    8952.0     10998.0       47337.0   
3  2022         4            Chaco   24582.0     61265.0       47285.0   
4  2022         4           Chubut   45450.0     73923.0        9815.0   

   Wireless    Otros      Total  
0  129910.0  72451.0  4794665.0  
1    4603.0  31826.0  1551926.0  
2    1379.0   2243.0    70909.0  
3    7957.0   2574.0   143663.0  
4   30721.0   8850.0   168759.0

In [231]:

print(df1['Trimestre'].value_counts())

Trimestre
4    216
3    216
2    216
1    216
Name: count, dtype: int64


In [229]:
df1.drop(df1.tail(1).index, inplace=True)

In [230]:
df1[['Trimestre','Año']] = df1[['Trimestre','Año']] .astype(int)
     


Queda el dataset df1 listo para poder usarlo.

## DF Accesos_a_banda_ancha_y_banda_angosta_por_provincia

In [232]:
df2 = pd.read_csv('Accesos_a_banda_ancha_y_banda_angosta_por_provincia.csv')
print(df2.info())
df2.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               864 non-null    int64  
 1   Trimestre         864 non-null    int64  
 2   Provincia         864 non-null    object 
 3   Banda ancha fija  864 non-null    int64  
 4   Dial up           862 non-null    float64
 5   Total             864 non-null    int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 40.6+ KB
None


Año  Trimestre        Provincia  Banda ancha fija  Dial up    Total
0  2022          4     Buenos Aires           4788466   6199.0  4794665
1  2022          4  Capital Federal           1549781   2145.0  1551926
2  2022          4        Catamarca             70908      1.0    70909
3  2022          4            Chaco            143658      5.0   143663
4  2022          4           Chubut            167855    904.0   168759

Analisis del DataFrame 'Accesos_banda_ancha_y_angosta_por_provincia'

vamos a verificar si tiene datos nulos.

In [234]:
datos_nulos = df2.isnull().sum()

# Imprimir los resultados
print(datos_nulos)

Año                 0
Trimestre           0
Provincia           0
Banda ancha fija    0
Dial up             2
Total               0
dtype: int64


- Hemos conseguido dos datos nulos en la columna Dial up

In [235]:
ubicacion_datos_nulos = df2[df2.isnull().any(axis=1)]

# Imprimir los resultados
print(ubicacion_datos_nulos)

      Año  Trimestre Provincia  Banda ancha fija  Dial up   Total
440  2018          2   Formosa             44554      NaN   44554
445  2018          2  Misiones            106147      NaN  106147


Vamos a dejar estas filas sin modificar ya que tiene infomración importante patra las otras columnas. 

De esta forma queda consolidado en df2.

In [236]:
df2[['Trimestre','Año']] = df2[['Trimestre','Año']] .astype(int)

##  DF Penetración_de _Internet fijo _(accesos _cada_100 hogares)

In [237]:
df3 = pd.read_csv('internet_Penetracion.csv')
print(df3.info())
df3.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       864 non-null    int64  
 1   Trimestre                 864 non-null    int64  
 2   Provincia                 864 non-null    object 
 3   Accesos por cada 100 hab  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB
None


Año  Trimestre        Provincia  Accesos por cada 100 hab
0  2022          4     Buenos Aires                     26.72
1  2022          4  Capital Federal                     50.35
2  2022          4        Catamarca                     16.73
3  2022          4            Chaco                     11.66
4  2022          4           Chubut                     26.23

## Analisis de df3, Penetración_de_Internet_fijo(accesos por cada 100 hogares)

- No observo valores Nulos.
- Cada una de las columnas tiene su respectivo formato.


In [240]:
df3.head()

Año  Trimestre        Provincia  Accesos por cada 100 hab
0  2022          4     Buenos Aires                     26.72
1  2022          4  Capital Federal                     50.35
2  2022          4        Catamarca                     16.73
3  2022          4            Chaco                     11.66
4  2022          4           Chubut                     26.23

In [241]:
datos_nulos = df3.isnull().sum()

# Imprimir los resultados
print(datos_nulos)

Año                         0
Trimestre                   0
Provincia                   0
Accesos por cada 100 hab    0
dtype: int64


In [242]:
df3.to_csv('Penetración_de_Internet_fijo(ac100 hogares).csv')

## DF historico_velocidad_media_Internet

In [248]:
df4 = pd.read_csv('historico_velocidad_bajada_internet.csv')
print(df4.info())
df4.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     864 non-null    int64  
 1   Trimestre               864 non-null    int64  
 2   Provincia               864 non-null    object 
 3   Mbps (Media de bajada)  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB
None


Año  Trimestre        Provincia  Mbps (Media de bajada)
0  2022          4     Buenos Aires                  111.00
1  2022          4  Capital Federal                  182.86
2  2022          4        Catamarca                   83.33
3  2022          4            Chaco                  104.02
4  2022          4           Chubut                   15.92

Tambien  pude extraer el archivo .csv de la media de velocidades por año.

In [245]:
df4_media = pd.read_csv('historico_velocidad_bajada_internet_media.csv')
print(df4.info())
df4_media.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     864 non-null    int64  
 1   Trimestre               864 non-null    int64  
 2   Provincia               864 non-null    object 
 3   Mbps (Media de bajada)  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB
None


Año  Trimestre  Mbps (Media de bajada)    Trimestre.1
0  2022          4                  106.56   Oct-Dic 2022
1  2022          3                   62.46  Jul-Sept 2022
2  2022          2                   58.44   Abr-Jun 2022
3  2022          1                   55.11   Ene-Mar 2022
4  2021          4                   52.34   Oct-Dic 2021

## DF Internet_Accesos-por-velocidad

In [251]:
df5 = pd.read_csv('Internet_Accesos-por-velocidad.csv')
print(df5.info())
df5.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  864 non-null    int64  
 1   Trimestre            864 non-null    int64  
 2   Provincia            864 non-null    object 
 3   HASTA 512 kbps       864 non-null    float64
 4   + 512 Kbps - 1 Mbps  864 non-null    float64
 5   + 1 Mbps - 6 Mbps    864 non-null    object 
 6   + 6 Mbps - 10 Mbps   864 non-null    float64
 7   + 10 Mbps - 20 Mbps  864 non-null    float64
 8   + 20 Mbps - 30 Mbps  864 non-null    float64
 9   + 30 Mbps            864 non-null    object 
 10  OTROS                858 non-null    float64
 11  Total                864 non-null    object 
dtypes: float64(6), int64(2), object(4)
memory usage: 81.1+ KB
None


Año  Trimestre        Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0  2022          4     Buenos Aires          28.934               23.347   
1  2022          4  Capital Federal         516.000                5.096   
2  2022          4        Catamarca          71.000              315.000   
3  2022          4            Chaco         461.000              771.000   
4  2022          4           Chubut         109.000                1.614   

  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0           259.866             281.550              244.555   
1            31.320              64.653               36.336   
2             2.716               5.028                4.638   
3            16.067              17.846                6.704   
4            46.457              31.007               34.728   

   + 20 Mbps - 30 Mbps  + 30 Mbps    OTROS      Total  
0               98.670  3.760.109   97.634  4.794.665  
1                8.116  1.299.117  106.772  1.551.926  
2                3.122     52.739    2.280     70.909  
3                3.039     94.866    3.909    143.663  
4               15.770     19.048   20.026    168.759

Analisis del DataFrame 'Internet_Accesos-por-velocidad'

In [71]:
datos_nulos = df5.isnull().sum()

# Imprimir los resultados
print(datos_nulos)

Año                    0
Trimestre              0
Provincia              0
HASTA 512 kbps         0
+ 512 Kbps - 1 Mbps    0
+ 1 Mbps - 6 Mbps      0
+ 6 Mbps - 10 Mbps     0
+ 10 Mbps - 20 Mbps    0
+ 20 Mbps - 30 Mbps    0
+ 30 Mbps              0
OTROS                  6
Total                  0
dtype: int64


- Podemos observar que tenemos 6 datos nulos en la columna otros , por los momentos no lo vamos a  tocar ya que esta columna no la valos a utilizar.
- Por otro lado podemos observar que las columnas  de 1 Mbps-6Mbps y +30 Mbps tienen formato tipo Object, vamos a cambiarla tipo Float.

In [72]:
df5['+ 1 Mbps - 6 Mbps'] = df5['+ 1 Mbps - 6 Mbps'].str.replace('.', '').astype(float)


In [75]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  864 non-null    int64  
 1   Trimestre            864 non-null    int64  
 2   Provincia            864 non-null    object 
 3   HASTA 512 kbps       864 non-null    float64
 4   + 512 Kbps - 1 Mbps  864 non-null    float64
 5   + 1 Mbps - 6 Mbps    864 non-null    float64
 6   + 6 Mbps - 10 Mbps   864 non-null    float64
 7   + 10 Mbps - 20 Mbps  864 non-null    float64
 8   + 20 Mbps - 30 Mbps  864 non-null    float64
 9   + 30 Mbps            864 non-null    object 
 10  OTROS                858 non-null    float64
 11  Total                864 non-null    object 
dtypes: float64(7), int64(2), object(3)
memory usage: 81.1+ KB


In [76]:
df5['+ 30 Mbps'] = df5['+ 30 Mbps'].str.replace('.', '').astype(float)

In [77]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  864 non-null    int64  
 1   Trimestre            864 non-null    int64  
 2   Provincia            864 non-null    object 
 3   HASTA 512 kbps       864 non-null    float64
 4   + 512 Kbps - 1 Mbps  864 non-null    float64
 5   + 1 Mbps - 6 Mbps    864 non-null    float64
 6   + 6 Mbps - 10 Mbps   864 non-null    float64
 7   + 10 Mbps - 20 Mbps  864 non-null    float64
 8   + 20 Mbps - 30 Mbps  864 non-null    float64
 9   + 30 Mbps            864 non-null    float64
 10  OTROS                858 non-null    float64
 11  Total                864 non-null    object 
dtypes: float64(8), int64(2), object(2)
memory usage: 81.1+ KB


In [184]:
df5.head()

Año  Trimestre        Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0  2022          4     Buenos Aires          28.934               23.347   
1  2022          4  Capital Federal         516.000                5.096   
2  2022          4        Catamarca          71.000              315.000   
3  2022          4            Chaco         461.000              771.000   
4  2022          4           Chubut         109.000                1.614   

   + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0           259866.0             281.550              244.555   
1            31320.0              64.653               36.336   
2             2716.0               5.028                4.638   
3            16067.0              17.846                6.704   
4            46457.0              31.007               34.728   

   + 20 Mbps - 30 Mbps  + 30 Mbps    OTROS      Total  
0               98.670  3760109.0   97.634  4.794.665  
1                8.116  1299117.0  106.772  1.551.926  
2                3.122    52739.0    2.280     70.909  
3                3.039    94866.0    3.909    143.663  
4               15.770    19048.0   20.026    168.759

vamos a crear un  data frame consolidado para su analisis  en el EDA

In [261]:
internet_T1= pd.merge(df1,df2, on=['Año','Trimestre','Provincia'])

In [262]:
internet_T1 = pd.merge(internet_T1, df3, on=['Año', 'Trimestre', 'Provincia'])

In [263]:
internet_T1 = pd.merge(internet_T1,df4, on=['Año', 'Trimestre', 'Provincia'])

In [264]:
internet_T1.drop('Total_x', axis=1, inplace=True)

In [265]:
internet_T1.head()

Año  Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2022          4     Buenos Aires  325908.0   2767247.0     1499149.0   
1  2022          4  Capital Federal  138753.0   1238196.0      138548.0   
2  2022          4        Catamarca    8952.0     10998.0       47337.0   
3  2022          4            Chaco   24582.0     61265.0       47285.0   
4  2022          4           Chubut   45450.0     73923.0        9815.0   

   Wireless    Otros  Banda ancha fija  Dial up  Total_y  \
0  129910.0  72451.0           4788466   6199.0  4794665   
1    4603.0  31826.0           1549781   2145.0  1551926   
2    1379.0   2243.0             70908      1.0    70909   
3    7957.0   2574.0            143658      5.0   143663   
4   30721.0   8850.0            167855    904.0   168759   

   Accesos por cada 100 hab  Mbps (Media de bajada)  
0                     26.72                  111.00  
1                     50.35                  182.86  
2                     16.73                   83.33  
3                     11.66                  104.02  
4                     26.23                   15.92

In [266]:
datos_nulos = internet_T1.isnull().sum()

# Imprimir los resultados
print(datos_nulos)

Año                         0
Trimestre                   0
Provincia                   0
ADSL                        0
Cablemodem                  0
Fibra óptica                0
Wireless                    0
Otros                       0
Banda ancha fija            0
Dial up                     2
Total_y                     0
Accesos por cada 100 hab    0
Mbps (Media de bajada)      0
dtype: int64


In [267]:
print(internet_T1[internet_T1['Dial up'].isnull()])

      Año  Trimestre Provincia     ADSL  Cablemodem  Fibra óptica  Wireless  \
440  2018          2   Formosa  23543.0      7311.0          18.0   13625.0   
445  2018          2  Misiones  57265.0     47923.0         512.0     391.0   

     Otros  Banda ancha fija  Dial up  Total_y  Accesos por cada 100 hab  \
440   57.0             44554      NaN    44554                  7.491908   
445   56.0            106147      NaN   106147                  8.615989   

     Mbps (Media de bajada)  
440                8.974987  
445               11.759657  


In [268]:
# Imputo el valor en nulos igual a 0
internet_T1['Dial up'].fillna(0, inplace=True) 

In [269]:
datos_nulos = internet_T1.isnull().sum()

# Imprimir los resultados
print(datos_nulos)

Año                         0
Trimestre                   0
Provincia                   0
ADSL                        0
Cablemodem                  0
Fibra óptica                0
Wireless                    0
Otros                       0
Banda ancha fija            0
Dial up                     0
Total_y                     0
Accesos por cada 100 hab    0
Mbps (Media de bajada)      0
dtype: int64


In [270]:
internet_T1.to_csv('internet_T_conexion.csv')

## Datasets: (Merge ['Provincia','Partido' 'Localidad'])
- Accesos_a_Internet_fijo_por_tecnologia_y_localidad
- Listado_de_localidades_con_conectividad_a_internet

Iniciamos con el dataframe Accesos_a_Internet_fijo_por_tecnologia_y_localidad.

In [271]:
df6 = pd.read_csv('AccesosaInternetfijoportecnologiaylocalidad_2791751698872938433.csv')
print(df6.info())
df6.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3137 entries, 0 to 3136
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Provincia      3137 non-null   object
 1   Partido        3137 non-null   object
 2   Localidad      3137 non-null   object
 3   Link Indec     3137 non-null   object
 4   ADSL           3137 non-null   int64 
 5   CABLEMODEM     3137 non-null   int64 
 6   DIAL UP        3137 non-null   int64 
 7   FIBRA OPTICA   3137 non-null   int64 
 8   OTROS          3137 non-null   int64 
 9   SATELITAL      3137 non-null   int64 
 10  WIMAX          3137 non-null   int64 
 11  WIRELESS       3137 non-null   int64 
 12  Total general  3137 non-null   int64 
dtypes: int64(9), object(4)
memory usage: 318.7+ KB
None


Provincia     Partido               Localidad Link Indec  ADSL  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo    6854100   874   
1  BUENOS AIRES  25 de Mayo               Del Valle    6854020   181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte    6854040     0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra    6854060     0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde    6854050     0   

   CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS  \
0        4542        0             3      0        852      0       706   
1           0        0             0     10          1      0         0   
2           0        0             0      0          0      0       181   
3         782        0             6    167          0      0       327   
4           0        0             0      0          0      0         6   

   Total general  
0           6977  
1            192  
2            181  
3           1282  
4              6

Analisis del DataFrame 'Accesos_a_Internet_fijo_por_tecnologia_y_localidad'
- No hay evidencia de datos nulos en este dataframe.
- Los nombres de provincia estan en mayusculas, los voy a cambiar .
- Hay algunos valores en las columnas numericas que tienen valores con '-0', hay que reemplazarlos por 0.
- Debo cambiar el tipo de dato de la columna ['ADSL','CABLEMODEM','DIAL UP','FIBRA OPTICA', 'OTROS',           'SATELITAL','WIMAX', 'WIRELESS' , 'Total general'] a float.
Hay una columna 'Link Indec' que no tiene relación con nuestro analisis, entonces la eliminare.
-

In [272]:
# Cambiar los nombre de las provicias por minuscula, manteniedo la primera en Mayuscula.
df6['Provincia'] = df6['Provincia'].str.capitalize()

In [273]:
# reemplazo de los valores de -0 por 0
df6.replace('- 0', '0', inplace=True)

In [274]:
numeric_data= ['ADSL','CABLEMODEM','DIAL UP','FIBRA OPTICA',
           'OTROS','SATELITAL','WIMAX', 'WIRELESS' , 'Total general' ]

In [276]:
#Elimino la columna Link Indec
df6.drop('Link Indec',axis=1, inplace=True)

In [277]:
df6.head()

Provincia     Partido               Localidad  ADSL  CABLEMODEM  \
0  Buenos aires  25 de Mayo              25 de Mayo   874        4542   
1  Buenos aires  25 de Mayo               Del Valle   181           0   
2  Buenos aires  25 de Mayo       Gobernador Ugarte     0           0   
3  Buenos aires  25 de Mayo  Norberto de la Riestra     0         782   
4  Buenos aires  25 de Mayo        Lucas Monteverde     0           0   

   DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS  Total general  
0        0             3      0        852      0       706           6977  
1        0             0     10          1      0         0            192  
2        0             0      0          0      0       181            181  
3        0             6    167          0      0       327           1282  
4        0             0      0          0      0         6              6

In [278]:
df6.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3137 entries, 0 to 3136
Data columns (total 12 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Provincia      3137 non-null   object
 1   Partido        3137 non-null   object
 2   Localidad      3137 non-null   object
 3   ADSL           3137 non-null   int64 
 4   CABLEMODEM     3137 non-null   int64 
 5   DIAL UP        3137 non-null   int64 
 6   FIBRA OPTICA   3137 non-null   int64 
 7   OTROS          3137 non-null   int64 
 8   SATELITAL      3137 non-null   int64 
 9   WIMAX          3137 non-null   int64 
 10  WIRELESS       3137 non-null   int64 
 11  Total general  3137 non-null   int64 
dtypes: int64(9), object(3)
memory usage: 294.2+ KB


 Continuamos ahora con el dataframe Listado_de_localidades_con_conectividad_a_internet

In [279]:
df7 = pd.read_csv('Listadodelocalidadesconconectividadainternet_2750221698873075984.csv')
print(df7.info())
df7.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4311 entries, 0 to 4310
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       4311 non-null   object 
 1   Partido         4311 non-null   object 
 2   Localidad       4311 non-null   object 
 3   Población       4311 non-null   int64  
 4   ADSL            4311 non-null   object 
 5   Cablemódem      4311 non-null   object 
 6   Dial Up         4311 non-null   object 
 7   Fibra óptica    4311 non-null   object 
 8   Satelital       4311 non-null   object 
 9   Wireless        4311 non-null   object 
 10  Telefonía Fija  4311 non-null   object 
 11  3G              4311 non-null   object 
 12  4G              4311 non-null   object 
 13  Link            4299 non-null   float64
 14  Latitud         4311 non-null   float64
 15  Longitud        4311 non-null   float64
dtypes: float64(3), int64(1), object(12)
memory usage: 539.0+ KB
None


Provincia     Partido                                Localidad  \
0  BUENOS AIRES  9 de Julio  Alfredo Demarchi (Est. Facundo Quiroga)   
1  BUENOS AIRES  9 de Julio                        Carlos María Naón   
2  BUENOS AIRES  9 de Julio                                 Dudignac   
3  BUENOS AIRES  9 de Julio                 La Aurora (Est. La Niña)   
4  BUENOS AIRES  9 de Julio           Manuel B. Gonnet (Est. French)   

   Población ADSL Cablemódem Dial Up Fibra óptica Satelital Wireless  \
0       1853   SI         --      --           --        --       --   
1        497   --         --      --           --        --       --   
2       2670   --         --      SI           --        --       SI   
3        531   --         --      --           --        --       --   
4        748   SI         --      --           SI        --       --   

  Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0             SI  SI  SI  6588010.0 -35.292982 -61.407254  
1             SI  SI  SI  6588020.0 -35.239499 -60.825136  
2             SI  SI  SI  6588040.0 -35.651193 -60.709896  
3             --  --  SI  6588050.0 -35.407947 -61.210567  
4             SI  SI  SI  6588060.0 -35.520141 -60.998949

In [280]:
df7['Provincia'] = df7['Provincia'].str.capitalize()

In [281]:
df7.head()

Provincia     Partido                                Localidad  \
0  Buenos aires  9 de Julio  Alfredo Demarchi (Est. Facundo Quiroga)   
1  Buenos aires  9 de Julio                        Carlos María Naón   
2  Buenos aires  9 de Julio                                 Dudignac   
3  Buenos aires  9 de Julio                 La Aurora (Est. La Niña)   
4  Buenos aires  9 de Julio           Manuel B. Gonnet (Est. French)   

   Población ADSL Cablemódem Dial Up Fibra óptica Satelital Wireless  \
0       1853   SI         --      --           --        --       --   
1        497   --         --      --           --        --       --   
2       2670   --         --      SI           --        --       SI   
3        531   --         --      --           --        --       --   
4        748   SI         --      --           SI        --       --   

  Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0             SI  SI  SI  6588010.0 -35.292982 -61.407254  
1             SI  SI  SI  6588020.0 -35.239499 -60.825136  
2             SI  SI  SI  6588040.0 -35.651193 -60.709896  
3             --  --  SI  6588050.0 -35.407947 -61.210567  
4             SI  SI  SI  6588060.0 -35.520141 -60.998949

## Inner Merge (['Provincia','Partido' 'Localidad'])
- Accesos_a_Internet_fijo_por_tecnologia_y_localidad
- Listado_de_localidades_con_conectividad_a_internet

In [282]:
internet_Loc= pd.merge(df6,df7, on=['Provincia','Partido','Localidad'])

In [283]:
internet_Loc.head()

Provincia     Partido               Localidad  ADSL_x  CABLEMODEM  \
0  Buenos aires  25 de Mayo              25 de Mayo     874        4542   
1  Buenos aires  25 de Mayo               Del Valle     181           0   
2  Buenos aires  25 de Mayo       Gobernador Ugarte       0           0   
3  Buenos aires  25 de Mayo  Norberto de la Riestra       0         782   
4  Buenos aires  25 de Mayo        Lucas Monteverde       0           0   

   DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  ...  Dial Up  Fibra óptica  \
0        0             3      0        852      0  ...       --            SI   
1        0             0     10          1      0  ...       --            --   
2        0             0      0          0      0  ...       --            --   
3        0             6    167          0      0  ...       --            SI   
4        0             0      0          0      0  ...       --            --   

   Satelital Wireless Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0         SI       SI             SI  SI  SI  6854100.0 -35.433939 -60.173121  
1         SI       --             --  SI  SI  6854020.0 -35.897345 -60.731609  
2         --       SI             --  --  --  6854040.0 -35.164397 -60.081358  
3         --       SI             SI  SI  SI  6854060.0 -35.272779 -59.771933  
4         --       SI             --  --  --  6854050.0 -35.483827 -59.988065  

[5 rows x 25 columns]

In [297]:
internet_Loc.to_csv('Mapa.csv')

In [284]:
# revisar los nulos y duplicados.
# Verify null values
print(internet_Loc.isnull().sum())

# Verify duplicated values
internet_Loc.duplicated().sum()

Provincia         0
Partido           0
Localidad         0
ADSL_x            0
CABLEMODEM        0
DIAL UP           0
FIBRA OPTICA      0
OTROS             0
SATELITAL         0
WIMAX             0
WIRELESS          0
Total general     0
Población         0
ADSL_y            0
Cablemódem        0
Dial Up           0
Fibra óptica      0
Satelital         0
Wireless          0
Telefonía Fija    0
3G                0
4G                0
Link              3
Latitud           0
Longitud          0
dtype: int64


0

no hay  valores duplicados 

In [287]:
print(internet_Loc.info())
internet_Loc.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2978 entries, 0 to 2977
Data columns (total 25 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       2978 non-null   object 
 1   Partido         2978 non-null   object 
 2   Localidad       2978 non-null   object 
 3   ADSL_x          2978 non-null   int64  
 4   CABLEMODEM      2978 non-null   int64  
 5   DIAL UP         2978 non-null   int64  
 6   FIBRA OPTICA    2978 non-null   int64  
 7   OTROS           2978 non-null   int64  
 8   SATELITAL       2978 non-null   int64  
 9   WIMAX           2978 non-null   int64  
 10  WIRELESS        2978 non-null   int64  
 11  Total general   2978 non-null   int64  
 12  Población       2978 non-null   int64  
 13  ADSL_y          2978 non-null   object 
 14  Cablemódem      2978 non-null   object 
 15  Dial Up         2978 non-null   object 
 16  Fibra óptica    2978 non-null   object 
 17  Satelital       2978 non-null   o

Provincia     Partido               Localidad  ADSL_x  CABLEMODEM  \
0  Buenos aires  25 de Mayo              25 de Mayo     874        4542   
1  Buenos aires  25 de Mayo               Del Valle     181           0   
2  Buenos aires  25 de Mayo       Gobernador Ugarte       0           0   
3  Buenos aires  25 de Mayo  Norberto de la Riestra       0         782   
4  Buenos aires  25 de Mayo        Lucas Monteverde       0           0   

   DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  ...  Dial Up  Fibra óptica  \
0        0             3      0        852      0  ...       --            SI   
1        0             0     10          1      0  ...       --            --   
2        0             0      0          0      0  ...       --            --   
3        0             6    167          0      0  ...       --            SI   
4        0             0      0          0      0  ...       --            --   

   Satelital Wireless Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0         SI       SI             SI  SI  SI  6854100.0 -35.433939 -60.173121  
1         SI       --             --  SI  SI  6854020.0 -35.897345 -60.731609  
2         --       SI             --  --  --  6854040.0 -35.164397 -60.081358  
3         --       SI             SI  SI  SI  6854060.0 -35.272779 -59.771933  
4         --       SI             --  --  --  6854050.0 -35.483827 -59.988065  

[5 rows x 25 columns]

## DF Ingresos_trimestrales_prestación_servicio_Internet_fijo

In [147]:
df8 = pd.read_csv('Ingresos_trimestrales_por_la_prestación_del_servicio.csv')
print(df8.info())
df8.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Año                        36 non-null     int64 
 1   Trimestre                  36 non-null     int64 
 2   Ingresos (miles de pesos)  36 non-null     int64 
 3   Periodo                    36 non-null     object
dtypes: int64(3), object(1)
memory usage: 1.3+ KB
None


Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2022          4                         72   Oct-Dic 2022
1  2022          3                         64  Jul-Sept 2022
2  2022          2                         60   Abr-Jun 2022
3  2022          1                         55   Ene-Mar 2022
4  2021          4                         45   Oct-Dic 2021

 ## Analisis del DataFrame 'Ingresos_trimestrales_prestación_servicio_Internet_fijo'
 - Puedo notar que no hay datos nulos.
 - Los datos  de la columna ['Ingresos (miles de pesos)'] esta en formato  int.
 - Los datos de la columna Año y trimestre estan en formato int. vamos a trabajar con este mismo formato.
 - Eliminar la columna 'Periodo' ya que tenemos la información del Trimestre y del Año, entonces es redundante.


In [ ]:
# Drop 'Periodo' column
df8.drop('Periodo', axis = 1, inplace = True)

Vamos a renombrar los dataSets que fueron trabajados e hicimos merge para usarlos en el EDA.

In [148]:
internet_ing = df8
internet_ran_v_prov = df5


## Nos quedan 4 dataset consolidados, que son:
internet_ing, internte_ran_v_prov, internet_loc, internet_T1

## Phone Datasets
## Fixed Phone Datasets

importamos es csv de telefonia fija.

In [292]:
df9 = pd.read_csv('Telefonia_Fija/Accesos_de_telefonía_fija_por_provincia_(hogares_comerciales_gobierno_otros).csv')
print(df9.info())
df9.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 8 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Año                                 864 non-null    int64  
 1   Trimestre                           864 non-null    int64  
 2   Provincia                           864 non-null    object 
 3   Accesos telefonía fija (total)      864 non-null    object 
 4   Accesos telefonía fija (hogares)    864 non-null    object 
 5   Accesos telefonía fija (comercial)  864 non-null    float64
 6   Accesos telefonía fija (gobierno)   864 non-null    float64
 7   Accesos telefonía fija (otros)      864 non-null    float64
dtypes: float64(3), int64(2), object(3)
memory usage: 54.1+ KB
None


Año  Trimestre        Provincia Accesos telefonía fija (total)  \
0  2022          4     Buenos Aires                      3.128.431   
1  2022          4  Capital Federal                      1.563.142   
2  2022          4        Catamarca                         25.083   
3  2022          4            Chaco                         64.683   
4  2022          4           Chubut                         54.249   

  Accesos telefonía fija (hogares)  Accesos telefonía fija (comercial)  \
0                        2.802.796                             264.038   
1                        1.400.437                             131.928   
2                           22.473                               2.117   
3                           57.950                               5.459   
4                           48.601                               4.579   

   Accesos telefonía fija (gobierno)  Accesos telefonía fija (otros)  
0                             12.029                          49.568  
1                              6.010                          24.767  
2                             96.000                         397.000  
3                            249.000                           1.025  
4                            209.000                         860.000

## Analisis del DataFrame 'Accesos_telefonía_fija_por_provincia-hogares-comercios-gobierno-otros'
 - No se perciben  datos nulos.
 - Las columnas Año, Trimestre y Provincia estan bien definidas.
 - Debo cambiar el tipo de dato de las columnas ['Accesos telefonía fija (hogares)','Accesos telefonía fija (total)'] a float.
 

In [289]:
df9['Accesos telefonía fija (total)'] = df9['Accesos telefonía fija (total)'].str.replace('.', '').astype(float)

In [290]:
df9['Accesos telefonía fija (hogares)'] = df9['Accesos telefonía fija (hogares)'].str.replace('.', '').astype(float)

In [291]:
print(df9.info())
df9.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 8 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Año                                 864 non-null    int64  
 1   Trimestre                           864 non-null    int64  
 2   Provincia                           864 non-null    object 
 3   Accesos telefonía fija (total)      864 non-null    float64
 4   Accesos telefonía fija (hogares)    864 non-null    float64
 5   Accesos telefonía fija (comercial)  864 non-null    float64
 6   Accesos telefonía fija (gobierno)   864 non-null    float64
 7   Accesos telefonía fija (otros)      864 non-null    float64
dtypes: float64(5), int64(2), object(1)
memory usage: 54.1+ KB
None


Año  Trimestre        Provincia  Accesos telefonía fija (total)  \
0  2022          4     Buenos Aires                       3128431.0   
1  2022          4  Capital Federal                       1563142.0   
2  2022          4        Catamarca                         25083.0   
3  2022          4            Chaco                         64683.0   
4  2022          4           Chubut                         54249.0   

   Accesos telefonía fija (hogares)  Accesos telefonía fija (comercial)  \
0                         2802796.0                             264.038   
1                         1400437.0                             131.928   
2                           22473.0                               2.117   
3                           57950.0                               5.459   
4                           48601.0                               4.579   

   Accesos telefonía fija (gobierno)  Accesos telefonía fija (otros)  
0                             12.029                          49.568  
1                              6.010                          24.767  
2                             96.000                         397.000  
3                            249.000                           1.025  
4                            209.000                         860.000

## DF Ingresos_trimestrales_telefonía fija.


In [159]:
df10 = pd.read_csv('Telefonia_Fija/Ingresos_trimestrales_por_la_prestación_del_servicio_de_telefonía fija.csv')
print(df10.info())
df10.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Año                    36 non-null     int64 
 1   Trimestre              36 non-null     int64 
 2   Ingresos (miles de $)  36 non-null     object
dtypes: int64(2), object(1)
memory usage: 996.0+ bytes
None


Año  Trimestre Ingresos (miles de $)
0  2014          1           291.607.175
1  2014          2           297.486.520
2  2014          3           310.546.202
3  2014          4           333.969.090
4  2015          1           326.012.475

## Analisis del DataFrame 'Ingresos_trimestrales_telefonía fija'
- Observamos que no  hay  evidencia de datos nulos
- Debo cambiar el tipo de dato de las columnas 'Ingresos (miles de $)'
a float



In [161]:
#cambiamos el formado de la columna Ingresos (miles de $) a float 
df10['Ingresos (miles de $)'] = df10['Ingresos (miles de $)'].str.replace('.', '').astype(float)

In [162]:
print(df10.info())
df10.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Año                    36 non-null     int64  
 1   Trimestre              36 non-null     int64  
 2   Ingresos (miles de $)  36 non-null     float64
dtypes: float64(1), int64(2)
memory usage: 996.0 bytes
None


Año  Trimestre  Ingresos (miles de $)
0  2014          1            291607175.0
1  2014          2            297486520.0
2  2014          3            310546202.0
3  2014          4            333969090.0
4  2015          1            326012475.0

Tenemos otro Dataset llamado Penetración_de_Internet_fijo(ac100 hogares).csv vamos  hacerle una revisión.

In [293]:
df10_1=  pd.read_csv('Penetración_de_Internet_fijo(ac100 hogares).csv')
print(df10_1.info())
df10_1.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                864 non-null    int64  
 1   Año                       864 non-null    int64  
 2   Trimestre                 864 non-null    int64  
 3   Provincia                 864 non-null    object 
 4   Accesos por cada 100 hab  864 non-null    float64
dtypes: float64(1), int64(3), object(1)
memory usage: 33.9+ KB
None


Unnamed: 0   Año  Trimestre        Provincia  Accesos por cada 100 hab
0           0  2022          4     Buenos Aires                     26.72
1           1  2022          4  Capital Federal                     50.35
2           2  2022          4        Catamarca                     16.73
3           3  2022          4            Chaco                     11.66
4           4  2022          4           Chubut                     26.23

In [296]:
df11 = pd.read_csv('Telefonia_Fija/Penetración_provincial_de_la_telefonía_fija_(accesos por cada 100 hogares).csv')
print(df11.info())
df11.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 4 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   Año                                          36 non-null     int64  
 1   Trimestre                                    36 non-null     int64  
 2   Accesos telefonía fija por cada 100 hab      36 non-null     float64
 3   Accesos telefonía fija por cada 100 hogares  36 non-null     float64
dtypes: float64(2), int64(2)
memory usage: 1.3 KB
None


Año  Trimestre  Accesos telefonía fija por cada 100 hab  \
0  2022          4                                15.455165   
1  2022          3                                15.456187   
2  2022          2                                15.566152   
3  2022          1                                15.969679   
4  2021          4                                16.740338   

   Accesos telefonía fija por cada 100 hogares  
0                                    49.743212  
1                                    49.786414  
2                                    50.179860  
3                                    51.520930  
4                                    53.879604

Analisis de Penetración_provincial_de_la_telefonía_fija_(accesos por cada 100 hogares)

- Observamos  que no tiene datos nulos.
- LAs unidades de cada una de las columnas tiene su formato correcto.

## Mobile Phone Datasets

## DF Ingresos_trimestrales_telefonía móvil


In [163]:
df11 = pd.read_csv('Telefonia_Movil/Telefonia_movil.csv')
print(df11.info())
df11.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 4 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Año                    40 non-null     int64 
 1   Trimestre              40 non-null     int64 
 2   Ingresos (miles de $)  40 non-null     object
 3   Periodo                40 non-null     object
dtypes: int64(2), object(2)
memory usage: 1.4+ KB
None


Año  Trimestre Ingresos (miles de $)        Periodo
0  2013          1          $ 13.135.235   Ene-Mar 2013
1  2013          2          $ 13.970.806   Abr-Jun 2013
2  2013          3          $ 15.235.659  Jul-Sept 2013
3  2013          4          $ 16.218.846   Oct-Dic 2013
4  2014          1          $ 15.938.353   Ene-Mar 2013

## Analisis del DataFrame 'Ingresos_trimestrales_telefonía móvil'
- No se observan  datos nulos.
- Se eliminan el simbolo dolar en la columna 'Ingresos (miles de $)'
- Debo cambiar el tipo de dato de las columnas 'Ingresos (miles de $)'a float.
- eliminación de la columna periodo ya que nos interesa por cada año.


In [165]:
# elimino el simbolo de $ en toda la columna Ingresos (miles de $)
df11['Ingresos (miles de $)'] = df11['Ingresos (miles de $)'].str.replace('$', '')

In [166]:
#cambiamos el formado de la columna Ingresos (miles de $) a float 
df11['Ingresos (miles de $)'] = df11['Ingresos (miles de $)'].str.replace('.', '').astype(float)

In [167]:
# elimino la columna Periodo.
df11.drop("Periodo", axis=1, inplace=True)

In [168]:
print(df11.info())
df11.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Año                    40 non-null     int64  
 1   Trimestre              40 non-null     int64  
 2   Ingresos (miles de $)  40 non-null     float64
dtypes: float64(1), int64(2)
memory usage: 1.1 KB
None


Año  Trimestre  Ingresos (miles de $)
0  2013          1             13135235.0
1  2013          2             13970806.0
2  2013          3             15235659.0
3  2013          4             16218846.0
4  2014          1             15938353.0

## DF Portabilidad_portaciones_netas_mensuales_por_operador.csv


In [169]:
df12 = pd.read_csv('Telefonia_Movil/Portabilidad_numerica_móvil_altas_mensuales_por_operador.csv')
print(df12.info())
df12.head

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130 entries, 0 to 129
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Año       130 non-null    int64  
 1   Mes       130 non-null    int64  
 2   Personal  130 non-null    float64
 3   Claro     130 non-null    float64
 4   Nextel    78 non-null     float64
 5   Movistar  130 non-null    float64
 6   Otros     10 non-null     float64
dtypes: float64(5), int64(2)
memory usage: 7.2 KB
None


Año  Mes  Personal    Claro  Nextel  Movistar  Otros
0  2022   12    46.782  113.363     NaN   109.610  224.0
1  2022   11    52.120  128.313     NaN   118.940  238.0
2  2022   10    60.855  117.430     NaN   119.716  153.0
3  2022    9    63.315  131.071     NaN   115.906  143.0
4  2022    8    68.659  132.676     NaN   115.879   42.0

In [171]:
num_nulls = df12['Nextel'].isnull().sum()
print(num_nulls)

52


In [172]:
num_nulls = df12.isnull().sum()
print(num_nulls)

Año           0
Mes           0
Personal      0
Claro         0
Nextel       52
Movistar      0
Otros       120
dtype: int64


 ## Analisis de Portabilidad_portaciones_netas_mensuales_por_operador
- exiten 52 datos nulos en la columna de Nextel.
- "Nextel, un proveedor de servicios de telecomunicaciones, actualmente no ofrece servicios de internet. Nextel se enfoca principalmente en servicios de comunicación móvil, como llamadas de voz y mensajería.
Teniendo esto en cuenta y los valores nulos, la columna relacionada con los servicios de internet de Nextel se puede eliminar ya que no es aplicable a nuestro análisis actual de Internet."
- existen 120 valores nulos en la columna  Otros. por tal motivo vamos a eliminar dicha columna  por no poseer  información relevante.


In [173]:
# elimino la columna Periodo.
df12.drop("Nextel", axis=1, inplace=True)

In [175]:
# elimino la columna Otros.
df12.drop("Otros", axis=1, inplace=True)

In [176]:
print(df12.info())
df12.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130 entries, 0 to 129
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Año       130 non-null    int64  
 1   Mes       130 non-null    int64  
 2   Personal  130 non-null    float64
 3   Claro     130 non-null    float64
 4   Movistar  130 non-null    float64
dtypes: float64(3), int64(2)
memory usage: 5.2 KB
None


Año  Mes  Personal    Claro  Movistar
0  2022   12    46.782  113.363   109.610
1  2022   11    52.120  128.313   118.940
2  2022   10    60.855  117.430   119.716
3  2022    9    63.315  131.071   115.906
4  2022    8    68.659  132.676   115.879

 ## DF Total_nacional_accesos_telefonia_movil-trimestral


In [177]:
df13 = pd.read_csv('Telefonia_Movil/Total_nacional_de_accesos(trimestral).csv')
print(df13.info())
df13.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 6 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Año                          40 non-null     int64 
 1   Trimestre                    40 non-null     int64 
 2   Total de accesos pospago     40 non-null     object
 3   Total de accesos prepago     40 non-null     object
 4   Total de accesos operativos  40 non-null     object
 5   Periodo                      40 non-null     object
dtypes: int64(2), object(4)
memory usage: 2.0+ KB
None


Año  Trimestre Total de accesos pospago Total de accesos prepago  \
0  2013          1                7.761.349               57.147.608   
1  2013          2                7.769.110               58.004.822   
2  2013          3                8.002.183               59.048.909   
3  2013          4                8.674.516               58.894.516   
4  2014          1                8.224.482               54.725.089   

  Total de accesos operativos        Periodo  
0                  64.908.957   Ene-Mar 2013  
1                  65.773.932   Abr-Jun 2013  
2                  67.051.092  Jul-Sept 2013  
3                  67.569.032   Oct-Dic 2013  
4                  62.949.571   Ene-Mar 2013

## Analisis de Total_nacional_accesos_telefonia_movil-trimestral
- No hay valores nulos.
- Se deben cambiar los '.' por '' en la columna ['Total de accesos pospago', 'Total de accesos prepago', 'Total de accesos operativos', 'Movistar'] y debo cambiar el tipo de dato de las columnas
a float.
- La columna 'Periodo' contiene los mismos datos de 'Año' y 'Trimestre', entonces es redundante y por eso la eliminare.

In [178]:
# Cambio de formato de la columna Total de accesos pospago
df13['Total de accesos pospago'] = df13['Total de accesos pospago'].str.replace('.', '').astype(float)

In [179]:
# Cambio de formato de la columna Total de accesos prepago
df13['Total de accesos prepago'] = df13['Total de accesos prepago'].str.replace('.', '').astype(float)

In [180]:
# Cambio de formato de la columna Total de accesos operativos
df13['Total de accesos operativos'] = df13['Total de accesos operativos'].str.replace('.', '').astype(float)

In [181]:
#Eliminación de la columna Periodo.
df13.drop("Periodo", axis=1, inplace=True)


In [182]:
print(df13.info())
df13.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Año                          40 non-null     int64  
 1   Trimestre                    40 non-null     int64  
 2   Total de accesos pospago     40 non-null     float64
 3   Total de accesos prepago     40 non-null     float64
 4   Total de accesos operativos  40 non-null     float64
dtypes: float64(3), int64(2)
memory usage: 1.7 KB
None


Año  Trimestre  Total de accesos pospago  Total de accesos prepago  \
0  2013          1                 7761349.0                57147608.0   
1  2013          2                 7769110.0                58004822.0   
2  2013          3                 8002183.0                59048909.0   
3  2013          4                 8674516.0                58894516.0   
4  2014          1                 8224482.0                54725089.0   

   Total de accesos operativos  
0                   64908957.0  
1                   65773932.0  
2                   67051092.0  
3                   67569032.0  
4                   62949571.0

## Penetración_nacional_de_la_telefonía_movil(accesos por cada 100 habitantes)

In [295]:
df14= pd.read_csv('Telefonia_Movil/Penetración_nacional_de_la_telefonía_móvil(accesos por cada 100 habitantes).csv')
print(df14.info())
df14.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       40 non-null     int64  
 1   Trimestre                 40 non-null     int64  
 2   Accesos por cada 100 hab  40 non-null     float64
 3   Periodo                   40 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB
None


Año  Trimestre  Accesos por cada 100 hab        Periodo
0  2013          1                    154.37   Ene-Mar 2013
1  2013          2                    156.00   Abr-Jun 2013
2  2013          3                    158.59  Jul-Sept 2013
3  2013          4                    159.37   Oct-Dic 2013
4  2014          1                    148.07   Ene-Mar 2014